eda_analysis/eda_analysis.py

```diff
@@ -78,8 +85,50 @@ def describe_cat_var(dataframe,cat_vars):
     >>> describe_cat_variable(X,cat_vars)
        
     """
-    # Code 
     
+    #Checking for valid inputs
+    if not isinstance(dataframe, pd.DataFrame):
+        raise Exception("The value of the argument 'dataframe' must be of type 'pandas.DataFrame'")
+    
+    if not isinstance(cat_vars,list) or not all(isinstance(x, str) for x in cat_vars):
+        raise Exception("The value of the argument 'cat_vars' must be a list of strings")
+    
+    dataframe=dataframe.dropna()
+    col_set = set(dataframe.columns)
+    col_subset = set(cat_vars)
+    if not col_subset.issubset(col_set):
+        raise Exception("The input categorical column names must belong to the dataframe")
+    
+
+    data = dataframe[cat_vars]
+    n = len(cat_vars)
+    n_cols = n_cols
+    n_rows = int(np.ceil(n/n_cols))
+    z = 0
+    
```

Comment:
Optional: you can melt the sub-dataframe and make the facet plot instead of using a for loop.
Otherwise, you may need to add tests to test every branch for the following if conditions.

Reply:
Yes I saw your code. I really liked your logic and have incorporated that in R. This I am not changing as it would be cumbersome to redo the code again. The tests I have written in such a way that it does all the coverage.

Reply:
Yes, I checked your line and branch coverage. All lines and branches are covered.